app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -279,6 +279,19 @@ export class MetamaskNotificationsController extends BaseController<
         },
       );
     },
+    initializePushNotifications: async () => {
+      if (!this.state.isMetamaskNotificationsEnabled) {
```

Comment:
Can notifications be toggled off per account? Does that need to be considered here?

Reply:
This is a great question!!

For context, we don't store any accounts directly. Instead we store "triggers". An account can have many triggers. E.g:
- trigger for send ERC20
- trigger for receive ERC20
- trigger for send ERC1155
- ...

Whenever we add or remove an account, we update/delete the relevant triggers, ensuring that we don't send push notifications for accounts that were disabled.